### PR TITLE
Kubernetes level CI test including AD

### PR DESCRIPTION
**Pull request by obnoxxx**
based on #42

### CODE REVIEW

**Comment by obnoxxx** [comment]
The ci check fails because the the dc can't be found when trying to join in the init container.
Manually putting a section for the domain into the coredns configmap works, but I couldn't find out how to automate it with kubectl patch, since in all my attempts it complains about '\n' characters in the json value... Still playing with it.

@phlogistonjohn

**Comment by obnoxxx** [comment]
@phlogistonjohn  - I figured it out. Now the patching of coredns works (probably it's possible more elegantly), and the joining and ad-member test work!

File: tests/files/samba-ad-server-deployment.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    matchLabels:
+      app: samba-ad
+  template:
+    metadata:
+      labels:
+        app: samba-ad
+    spec:
+      containers:
+      - name: samba
+        # Pointing to locally built image,
```
Comment (phlogistonjohn) [inline tests/files/samba-ad-server-deployment.yml:19]: Comment doesn't seem right to me.

**Review by phlogistonjohn — CHANGES_REQUESTED** [review]
Looks awesome, but I do see one tiny issue I think we should fix as it could be confusing in the future. Thanks!

**Comment by phlogistonjohn** [comment]
> I figured it out. Now the patching of coredns works (probably it's possible more elegantly), and the joining and ad-member test work!

I spent some time on last friday looking for such a more elegant way, and I didn't find one, so this is pretty fine IMO.

File: tests/files/samba-ad-server-deployment.yml (hunk 1 above)
Comment (obnoxxx) [reply tests/files/samba-ad-server-deployment.yml]: correct. fixed :-)

**Pull request merged**